website/docs/feature-flag-tutorials/rust/implementing-feature-flags-rust.md

```diff
@@ -0,0 +1,286 @@
+cargo add serde --features derive
+cargo add tokio --features full
+cargo add unleash-api-client --features reqwest
+```
+
+There are a few dependencies, and here's why: We need an HTTP client to make the request, and `serde` to deserialize the Unleash response. Our SDK is constantly polling Unleash to retrieve the value of feature flags, and caches that in memory. We don't want to block your app while this happens, so we need an async runtime. We're picking `tokio`.
+
+```rust
+use enum_map::Enum;
+use image::io::Reader as ImageReader;
+use ravif::{Encoder as AvifEncoder, Img};
+use rgb::FromSlice;
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use tokio::time::sleep;
+use unleash_api_client::client;
+use webp::Encoder as WebPEncoder;
+
+#[derive(Debug, Deserialize, Serialize, Enum, Clone)]
+enum Flags {
+    avif,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    let client = client::ClientBuilder::default().into_client::<Flags, reqwest::Client>(
+        "http://localhost:4242/api",
+        "unleash-rust-client-example",
+        "unleash-rust-client-example",
+        Some("default:development.unleash-insecure-api-token".to_string()),
+    )?;
+    client.register().await?;
+
+    let (_, _) = tokio::join!(client.poll_for_updates(), async {
+        // Ensure we have features for this demo.
+        sleep(Duration::from_millis(500)).await;
+
+        let avif = client.is_enabled(Flags::avif, None, false);
+        process_images(avif)?;
+
+        // allow tokio::join to finish
+        client.stop_poll().await;
+        Ok::<(), Box<dyn std::error::Error + Send + Sync + 'static>>(())
+    });
+
+    Ok(())
+}
+
+fn process_images(
+    convert_to_avif: bool,
+) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    let img = ImageReader::open("input.png")?.decode()?;
+
+    if convert_to_avif {
+        // Convert to AVIF
+        let rgba_image = img.to_rgba8();
+        let pixels_rgba = rgba_image.as_raw().as_rgba();
+        let avif_data = AvifEncoder::new().with_quality(75.0).encode_rgba(Img::new(
+            pixels_rgba,
+            rgba_image.width() as usize,
+            rgba_image.height() as usize,
+        ))?;
+        std::fs::write("output.avif", avif_data.avif_file)?;
+        println!("Image conversion to AVIF completed successfully.");
+    } else {
+        // Convert to WebP
+        let webp_data = WebPEncoder::from_image(&img)?.encode(75.0);
+        std::fs::write("output.webp", webp_data.to_vec())?;
+        println!("Image conversion to WebP completed successfully.");
+    }
+
+    Ok(())
+}
+```
+
+## 5. Verify the toggle experience
+
+Now that we’ve connected our project to Unleash and grabbed our feature flag, we can verify that if you disable that flag in your development environment, you should only see the WebP conversion.
+
+> **Note:** An update to a feature flag may take 30 seconds to propagate.
```

Comment:
This is within your control - you're setting up the client in the builder, you can just set this to a second for this tutorial (I think that's a nice idea anyway so long as you warn folks it's for demo reasons).

The poll interval in the SDK SDK seems to be 15 seconds though, so, updates shouldn't take longer than that

Reply:
Agree with Simon's points here. Setting the poll rate lower for a tutorial seems sensible (and makes it more fun to follow along with when you don't have to wait).

Also, this should be a docusaurus admonition again. Maybe something like this (though it feels a bit **too** wordy (even for me 😅 ))


```suggestion
:::info
Your feature flag configuration will only update as often as your SDK polls Unleash. The default polling interval for the Rust SDK is 15 seconds, but you can set whatever interval works best for your use case.
:::
```